### DISCUSSION TITLE
Queue-based scaling is killing container instances before completion (Azure Container Apps)

### DISCUSSION THREAD

**Discussion by Jasenka2** [microsoft/azure-container-apps#725] [Q&A]
I am running container which is processing long running task in Azure Container Apps environment.
I setup Azure Queue storage Scaler with queue length 1 because I want that with each new message on queue new replica spin up. New instance will receive first available (visible) message on queue, lock the message, do processing based on message information and when processing is finished delete the message from queue. Scale up of containers works fine, but after ~5 minutes containers which have not finished processing and intensively used memory (so container is not in any idle state in terms of resources) are just scale down (shut down) by scaler.
Is there any recommendation how to resolve this? I did not find way how to reconfigure scaler limits for scaling down.
Is there some other better solution for controlling scaling of long running containers in ACA environment?

**Comment by BrianVallelunga** [comment]
This is hitting us as well. We have a container app that takes a message from a queue and then processes it. That processing may take 10 minutes and in the time the app may be scaled down.

I'd like some way of signaling to the host that the app is currently working. This could be done with an endpoint, or some other method. As it is, this makes the scaling rather useless for anything that's not a direct request/response.
